intake/catalog/entry.py

```diff
@@ -125,6 +126,28 @@ def plots(self):
         """List custom associated quick-plots """
         return list(self._metadata.get('plots', {}))
 
+    def _ipython_display_(self):
+        """Display the entry as a rich object in an IPython session."""
+        contents, warning = self._display_content()
+        display({ # noqa: F821
+            'application/json': contents,
+            'text/plain': pretty_describe(contents)
+        }, metadata={
+            'application/json': { 'root': self.name }
```

Comment:
```suggestion
            'application/json': { 'root': self._name }
```
You'll want to use the underscore name to avoid initializing the entry I think

Reply:
Do you think we should have a test driver specifically to guard against this?

```
class DoNotInitialise(Source):

    def __init__(self, *args, **kwargs):
        raise RuntimeError
```

Reply:
I see that `_name` is defined on `LocalCatalogEntry` and `name` is defined on `RemoteCatalogEntry`, and I don't see either on the base class. Is it safe to use `_name` here?

Reply:
It should consistently be `_name`. I don't know how the attributes all ended up underscored for the entries, but I suppose we should stick with it for now.

Reply:
I suppose I am missing some context/motivation here. For my own edification, what is the reason to avoid initializing?

Reply:
It can take some non-zero time. In theory, sources ought to defer any serious processing until at least `.discover()` (i.e., inferring the schema and other important details) but in practice some stuff can happen in the driver `__init__` too.